### PR TITLE
feat(mysql): add re_extract

### PR DESCRIPTION
**Pull request by mesejo**

### CODE REVIEW

File: ibis/backends/mysql/tests/conftest.py [hunk 1]
```diff
@@ -144,3 +144,18 @@ def con_nodb():
...
+
+
+@pytest.fixture(scope='module')
+def db(con):
+    return con.database()
+
+
+@pytest.fixture(scope='module')
+def alltypes(db):
+    return db.functional_alltypes
```
Comment (cpcloud) [inline ibis/backends/mysql/tests/conftest.py:156]:
```suggestion
@pytest.fixture(scope='module')
def alltypes(con):
    return con.tables.functional_alltypes
```

**Review by cpcloud — APPROVED** [review]
LGTM, one small nit and a question!

File: ibis/backends/mysql/tests/test_string.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        ),
+        param(
+            lambda t: t.date_string_col.re_extract(r'^(\d+)', 1),
+            lambda t: t.date_string_col.str.extract(r'^(\d+)', expand=False),
+            id='re_extract_group_at_beginning',
+        ),
+        param(
+            lambda t: t.date_string_col.re_extract(r'(\d+)$', 1),
+            lambda t: t.date_string_col.str.extract(r'(\d+)$', expand=False),
+            id='re_extract_group_at_end',
```
[the remark this answers is not in the document]
Comment (mesejo) [reply ibis/backends/mysql/tests/test_string.py:38]: Not really. They should work, for every backend.

[the remark this answers is not in the document]
Comment (mesejo) [reply ibis/backends/mysql/tests/test_string.py:38]: Should I add them in the list of tests for all backends?

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/mysql/tests/test_string.py:38]: Yeah, that'd be swell!

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/mysql/tests/test_string.py:38]: I will merge this PR though, and it's fine to move these tests into the broader suite in a follow-up!

**Pull request merged**